doc/dev/debugging.rst

```diff
@@ -0,0 +1,279 @@
+* ``Unable to link... All available compiler options exhausted ...``: Even though Catalyst has a list of compilers that may be available in your system, it is possible that no compiler successfully linked the program to the libraries shipped with Catalyst.
+  This should be a rare case, but if you ever find yourself seeing this error, you should check that one of the following compilers or interfaces are available in your ``PATH``: ``clang``, ``gcc``, ``c99``, ``c89``, ``cc``.
+  Correctly installing any of these should be sufficient.
+  Alternatively, you can also specify a preferred compiler to use during linking using the environment variable ``CATALYST_CC``.
+  For example ``CATALYST_CC=clang-14`` specifies that ``clang-14`` will be the preferred compiler to be used during the linking stage.
+* ``User defined compiler is not in path``: Following the example above, you may have installed ``clang-14`` in a different directory that is not currently on the ``PATH``.
+  Please make sure that the executable specified in ``CATALYST_CC`` is available in the ``PATH`` and is flag compatible with ``gcc`` or ``clang``.
+
+Running without Python runtime
+==============================
+
+Sometimes, it might be useful to run the JIT compiled function without Python, for example in case of debugging.
+If the program succeeds without Python, then it is likely that the error is found in the interface between Python and the JIT compiled function, or within Catalyst's Python internals.
+As a quick sanity check you might want to run the JIT compiled function without Python.
+Below is an example of how to obtain a C program that can be linked against the generated function:
+
+.. code-block:: python
+
+    @qjit
+    def identity(x):
+        return x
+
+    print(identity.get_cmain(1.0))
+
+Using the ``QJIT.get_cmain`` function, the following string is returned to the user:
+
+.. code-block:: C
+
+    #include <complex.h>
+    #include <stddef.h>
+    #include <stdint.h>
+
+    typedef int64_t int64;
+    typedef double float64;
+    typedef float float32;
+    typedef double complex complex128;
+    typedef float complex complex64;
+
+
+    struct memref_float64x0_t
+    {
+        float64* allocated;
+        float64* aligned;
+        size_t offset;
+
+    };
+    struct result_t {
+        struct memref_float64x0_t f_0;
+    };
+
+
+    extern void setup(int, char**);
+    extern void _catalyst_ciface_jit_identity(struct result_t*, struct memref_float64x0_t*);
+    extern void teardown();
+
+    int
+    main(int argc, char** argv)
+    {
+
+        struct result_t result_val;
+        float64 buff_0 = 1.0;
+        struct memref_float64x0_t arg_0 = { &buff_0, &buff_0, 0 };
+
+
+        setup(1, &argv[0]);
+        _catalyst_ciface_jit_identity(&result_val, &arg_0);
+        teardown();
+    }
+
+The user can now compile and link this program and run it without Python.
+
```

Comment:
That would be nice to mention verbosity modes in Cataylst as well.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/33/commits/228d6f8cb20cd57c53bc820b21f8df0c86e0db78